backend-zs/middleware/auth.js

```diff
@@ -0,0 +1,16 @@
+const jwt = require("jsonwebtoken");
+
+module.exports = {
+    authenticateUser: function (req, res, next) {
```

Comment:
@dimitrityan I added the middleware to decode the jwt token for endpoints that would only be accessed by logged in users. Decoded info will be found at req.user. Any comments on the implementation? Is this how you are planning to do it?